crates/solver/src/settlement.rs

```diff
@@ -299,6 +303,78 @@ impl Settlement {
         }
     }
 
+    // Checks whether the settlement prices do not deviate more than max_settlement_price_deviation from the auction prices on certain pairs.
+    pub fn satisfies_price_checks(
+        &self,
+        auction_id: u64,
+        solver_name: &str,
+        external_prices: &ExternalPrices,
+        max_settlement_price_deviation: &Ratio<BigInt>,
+        tokens_to_satisfy_price_test: &Option<HashSet<H160>>,
```

Comment:
Maybe a way to make this simpler would be to:
```suggestion
        tokens_to_satisfy_price_test: &HashSet<H160>,
```

Since `None` and `Some(HashSet::default())` are equivalent? Then we can always `unwrap_or_default()` when creating it so we don't have to handle `None` values everywhere.

Reply:
I wanted to have this optional, such that I have an option to run the test on all tokens.

I think I should name the variable differently. Probably: `restrictive_token_set_for_price_check`. Then it would be much easier to understand!  I will change it.

Reply:
For None, I was imaging it that we would test on all tokens, and for an empty set, we would test it on no tokens.

Probably, I could have that made clear by a better variable name: 'restrictive_token_list_for_price_check'? Then none means no restriction on the token set, and empty means that we have a very strong restriction - name no tokens?
Would that make it easier to understand?

Reply:
Same confusion as ☝️